ext/msgpack/factory_class.c

```diff
@@ -68,15 +69,24 @@ static VALUE Factory_alloc(VALUE klass)
 static VALUE Factory_initialize(int argc, VALUE* argv, VALUE self)
 {
     FACTORY(self, fc);
+    VALUE options = Qnil;
 
     msgpack_packer_ext_registry_init(&fc->pkrg);
     msgpack_unpacker_ext_registry_init(&fc->ukrg);
 
     fc->has_symbol_ext_type = false;
+    fc->strict_types = false;
 
     switch (argc) {
     case 0:
         break;
+    case 1:
+        options = argv[0];
+        if(rb_type(options) != T_HASH) {
+            rb_raise(rb_eArgError, "expected Hash but found %s.", rb_obj_classname(options));
+        }
+        fc->strict_types = rb_hash_aref(options, ID2SYM(rb_intern("strict")));
```

Comment:
It would be preferable to parse keyword arguments rather than take an options hash.